### PR TITLE
Added Scheduling Commands POC Feature

### PR DESCRIPTION
**Pull request by quinnha**
Added scheduling functionality to schedule commands for POC

Scheduling reads from available commands in the satellite (TO-DO: restructure command structure, potentially in a DB). Currently scheduling console logs commands sent, with `sendSchedule()` to be modified with actual logic on scheduling commands, potentially batching in a queue.

Also added `getAllSatellites` endpoint for helper function

https://github.com/RishiVaya/Lower_Earth_Orbiters/assets/35006641/35ece1b8-ccdc-4352-b518-9da6e86d7fff

Note: for some reason the package stuff all changed too which is not nice :(

### CODE REVIEW

**Comment by quinnha** [comment]
forgot to demo clear schedule but trust me bro it works

**Comment by RishiVaya** [comment]
when u do send schedule, do we not want it to auto clear, so u can work on next schedule's ccommands?

**Comment by quinnha** [comment]
> when u do send schedule, do we not want it to auto clear, so u can work on next schedule's ccommands?

fixed

File: src/leo-client-app/src/components/Scheduler.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
+"use client";
```
Comment (RishiVaya) [inline src/leo-client-app/src/components/Scheduler.tsx:1]: is this necessary?

File: src/leo-client-app/src/pages/globals.css [hunk 2]
```diff
@@ -109,3 +109,28 @@ a {
...
+  cursor: pointer;
+  width: 100%;
+  border-radius: 8px;
+}
+
+.removeButton {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
```
Comment (RishiVaya) [inline src/leo-client-app/src/pages/globals.css:127]: can this just be extended?

**Review by RishiVaya — APPROVED** [review]
lgtm

File: src/leo-server-app/src/routes/satellite.ts
Comment (Umang-Rajkarnikar) [inline src/leo-server-app/src/routes/satellite.ts:1]: Lgtm, just wondering why change to query?

**Review by Umang-Rajkarnikar — APPROVED** [review]

File: src/leo-client-app/src/pages/manage-operators/index.tsx [hunk 3]
```diff
@@ -2,12 +2,10 @@
 
 import Navbar from "@/components/navbar/Navbar";
 import OperatorList from "@/components/operator-management/OperatorList";
-import { useUser, withPageAuthRequired } from "@auth0/nextjs-auth0/client";
+import { withPageAuthRequired } from "@auth0/nextjs-auth0/client";
 import { Stack } from "@mui/material";
 
 function MangeOperatorsPage() {
-  const { user } = useUser();
-
```
Comment (RishiVaya) [inline src/leo-client-app/src/pages/manage-operators/index.tsx:10]: why did u remove this?

File: src/leo-client-app/src/components/Scheduler.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      })
+      .catch((error) => {
+        console.error("Error fetching valid commands:", error);
+      }); 
+  };
+
+  const addCommand = (command : string) => {
+    setCurrentSchedule(prevCommands => [...prevCommands, command]);
+  };
+
```
Comment (RishiVaya) [inline src/leo-client-app/src/components/Scheduler.tsx:26]: general requirement question - do we need some validation here (eg - stacking same commands, too many commands, looping?)

**Review by RishiVaya — COMMENTED** [review]
this review better?

File: src/leo-client-app/src/components/Scheduler.tsx (hunk 4 above)
Comment (quinnha) [reply src/leo-client-app/src/components/Scheduler.tsx:26]: not sure where we could add validation here, the most we have is just if the commands exist or not? this is just getting all the possible commands for a satellite

File: src/leo-client-app/src/pages/manage-operators/index.tsx (hunk 3 above)
Comment (quinnha) [reply src/leo-client-app/src/pages/manage-operators/index.tsx:10]: not being used

File: src/leo-server-app/src/routes/satellite.ts
Comment (quinnha) [reply src/leo-server-app/src/routes/satellite.ts:1]: so it actually works when i call it thru the code, couldnt get it to work if i didn't do this

**Comment by quinnha** [comment]
yep @RishiVaya much better

**Pull request merged**